src/OpenTelemetry.Instrumentation.EntityFrameworkCore/Implementation/EntityFrameworkDiagnosticListener.cs

```diff
@@ -92,7 +92,18 @@ public override void OnEventWritten(string name, object? payload)
                         var dataSource = (string)this.dataSourceFetcher.Fetch(connection);
                         if (!string.IsNullOrEmpty(dataSource))
                         {
-                            this.AddTag(activity, ("peer.service", SemanticConventions.AttributeServerAddress), dataSource);
+                            var connectionDetails = SqlConnectionDetails.ParseFromDataSource(dataSource);
+
+                            var serverAddress = connectionDetails.ServerHostName ?? connectionDetails.ServerIpAddress;
+                            if (!string.IsNullOrEmpty(serverAddress))
+                            {
+                                this.AddTag(activity, ("peer.service", SemanticConventions.AttributeServerAddress), serverAddress);
```

Comment:
Note part of this PR, but it is not part of https://github.com/open-telemetry/semantic-conventions/blob/v1.36.0/docs/database/database-spans.md, probably never was...

Reply:
Yeah, I spotted this when I did a previous refactor PR, but figured it would break people who'd depended on it, so wasn't sure what to do.

I figure worst case, it just gets removed when the library is stable and all the old Sem Convs are removed.

Reply:
Yes, but we need to do this, before stabilization